### PR TITLE
[14][IMP] shopfloor_delivery_shipment improve confirmation of shipment creation

### PR DESCRIPTION
**Pull request by TDu**

### CODE REVIEW

**Review by jbaudoux — APPROVED** [review]

File: shopfloor_delivery_shipment/services/delivery_shipment.py [hunk 1]
```diff
@@ -68,7 +68,12 @@ def scan_dock(self, barcode, confirmation=False):
                 if not confirmation:
                     return self._response_for_scan_dock(
                         message=self.msg_store.scan_dock_again_to_confirm(dock),
-                        confirmation_required=True,
+                        confirmation_required=dock.barcode,
```
Comment (jbaudoux) [inline shopfloor_delivery_shipment/services/delivery_shipment.py]:
```suggestion
                        confirmation_required=barcode,
```

File: shopfloor_delivery_shipment/services/delivery_shipment.py [hunk 2]
```diff
@@ -68,7 +68,12 @@ def scan_dock(self, barcode, confirmation=False):
...
+                    )
+                if confirmation != dock.barcode:
```
Comment (jbaudoux) [inline shopfloor_delivery_shipment/services/delivery_shipment.py]:
```suggestion
                if confirmation != barcode:
```

File: shopfloor_delivery_shipment/services/delivery_shipment.py [hunk 3]
```diff
@@ -68,7 +68,12 @@ def scan_dock(self, barcode, confirmation=False):
...
+                    return self._response_for_scan_dock(
+                        message=self.msg_store.scan_dock_again_to_confirm(dock),
+                        confirmation_required=dock.barcode,
```
Comment (jbaudoux) [inline shopfloor_delivery_shipment/services/delivery_shipment.py]:
```suggestion
                        confirmation_required=barcode,
```

**Review by jbaudoux — CHANGES_REQUESTED** [review]

**Comment by TDu** [comment]
Rebased, did the changes requested

**Review by jbaudoux — APPROVED** [review]

**Pull request merged**